src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs

```diff
@@ -78,5 +93,41 @@ public void Create_Encrypted_RouteString_From_Dictionary()
             
             Assert.AreEqual(expectedResult, result);
         }
+
+        [Test]
+        public void Truncate_By_Words()
+        {
+            var text = "Hello world, this is some text <a href='blah'>with a link</a>";
+
+            var helper = new UmbracoHelper();
+
+            var result = helper.TruncateByWords(text, 4).ToString();
+
+            Assert.AreEqual("Hello world, this is&hellip;", result);
+        }
+
+        [Test]
+        public void Truncate_By_Words_With_Tag()
+        {
+            var text = "Hello world, <b>this</b> is some text <a href='blah'>with a link</a>";
```

Comment:
Try changing this to `Hello world, <b>this is some</b>`. You'll notice it will not work because it is pretty hard to do this correctly with HTML in the mix. 

I don't know why we'd want this in the core, there's libraries out there to help with this, so we don't have to reinvent the wheel, example: https://robvolk.com/truncate-html-string-c-extension-method-51c83e6d4969

And proving that it is pretty difficult, he still has an open issue: https://github.com/robvolk/Helpers.Net/issues/3

Also note that it is entirely possible that I put invalid HTML in there, so something like `<p>testing <b>word truncation</p>` - the `</b>` is missing, that should cause some "fun" things to happen too.

If we want this in core, I'd suggest we create a version that strips all HTML so we don't run the risk of breaking the page this is used on.

Reply:
Hey @nul800sebastiaan , about the "Hello world, <b>this is some</b>", I was pretty sure that it worked as I am stripping the HTML elements from the string, convert it to char length and feed it to the original truncate method. I will look into it again today.

About invalid tags, funny enough I ran the string you provided and the Truncate method closes the tags correctly, probably not as you wanted but... at least is not invalid HTML. Talking about invalid HTML, should we worry about that? I mean, if you type invalid HTML it should brake.

Oh, and when you tried all that, was "treatTagsAsContent" true or false? Either way, I'll go trough another round of testing today, thanks for your input :).